Command/PopulateCommand.php

```diff
@@ -109,20 +110,34 @@ protected function execute(InputInterface $input, OutputInterface $output)
      */
     private function populateIndex(OutputInterface $output, $index, $reset, $options)
     {
+
+        /** @var $providers ProviderInterface[] */
+        $providers = $this->providerRegistry->getIndexProviders($index);
+
         if ($reset) {
             $output->writeln(sprintf('<info>Resetting</info> <comment>%s</comment>', $index));
             $this->resetter->resetIndex($index, true);
         }
 
-        /** @var $providers ProviderInterface[] */
-        $providers = $this->providerRegistry->getIndexProviders($index);
-
         foreach ($providers as $type => $provider) {
-            $loggerClosure = function($message) use ($output, $index, $type) {
-                $output->writeln(sprintf('<info>Populating</info> %s/%s, %s', $index, $type, $message));
-            };
+            if (class_exists('Symfony\Component\Console\Helper\ProgressBar')) {
+                $output->writeln(sprintf('<info>Populating</info> %s/%s', $index, $type));
+                $progressBar = new ProgressBar($output, $provider->getTotalObjects());
+                $progressBar->setFormat('debug');
+                $progressBar->start();
+                $loggerClosure = function($number) use ($progressBar) {
+                    $progressBar->advance($number);
+                };
+            } else {
+                $loggerClosure = function($message) use ($output, $index, $type) {
```

Comment:
use the ProgressHelper when the ProgressBar is not there (it will allow to share most logic, and to have a consistent output)